src/Workspaces/Remote/ServiceHub/Shared/ServiceHubServiceBase.cs

```diff
@@ -82,6 +91,37 @@ protected RoslynServices RoslynServices
             }
         }
 
+        protected bool IsDisposed => ((IDisposableObservable)_rpc).IsDisposed;
```

Comment:
Should this be tied to _disposed?

Reply:
I can change name to IsRpsDisposed. if that make it not confusing.

Reply:
Isn't it the case that Rpc object will only be disposed is if this type itself is disposed? IsRpcDisposed may be confusing if Rpc is a private implementation detail of this type...